### DISCUSSION TITLE
All new deployments fail in 504 FUNCTION_INVOCATION_TIMEOUT

### DISCUSSION THREAD

**Discussion by szhshp** [vercel/community#1813] [Help] [answered]
### Summary

Hey Vercel team,

Recently, I'm not sure what happened. I haven't made any changes to the dependencies.

I'm having trouble with a new deployment in my project - it's giving me a 504 timeout error, and all my previous deployments are showing the same issue.

I had a similar problem before and fixed it by changing `{ "node": "16.x" }`, but that's not working this time.

Could you help me take a look and see what's going on?

The final Deployment: [504: GATEWAY_TIMEOUT ](https://szhshp-remix-hbg1bj1o1-szhshp.vercel.app/)

![image](https://user-images.githubusercontent.com/10084666/226113940-4b39f872-c811-4910-aa74-309893e1a06a.png)

### Additional information

```jsx
504: GATEWAY_TIMEOUT
Code: FUNCTION_INVOCATION_TIMEOUT
ID: hnd1::fs8k9-1679151302268-044a719e1f27
```

### Example

https://szhshp-remix-hbg1bj1o1-szhshp.vercel.app/

**Comment by amyegan** [comment, marked as answer]
Thank you for reaching out. The most common reason for a [FUNCTION_INVOCATION_TIMEOUT Error](https://vercel.com/docs/error/application/FUNCTION_INVOCATION_TIMEOUT) is that a Serverless Function reached the [timeout limit](https://vercel.com/docs/concepts/limits/overview#serverless-function-execution-timeout) (10 seconds for Hobby accounts, 60 seconds for Pro accounts, and 90 seconds for Enterprise accounts) because of an upstream provider taking too long to respond.

The first step I would typically recommend is to check your upstream provider for slowness. If you already confirmed that isn't the problem, do you have a [minimal reproducible example](https://vercel.com/support/articles/creating-a-minimal-reproducible-example)  that we can use for debugging?

**Reply by jessepinho** [reply]
Hey @amyegan, I'm experiencing the exact same issue as OP. I can't figure it out, and the timeouts are not due to some upstream dependency taking a while. Also, same as OP, this change seemed to happen out of the blue, not due to some code change I made.

I've created a minimal reproducible example:

- Edge function (works fine): https://dev.film.jessepinho.com/api/test-edge-function (source)
- Serverless function (times out): https://dev.film.jessepinho.com/api/test-serverless-function
- Source for both functions: https://github.com/jessepinho-film-tools/film-tools-web/commit/cc78ddcb2675519dd370e598027cd107e038eabf#diff-815bea3065b82481dd5ee16895173dddbcd0efcb632077bcf2a0d954809077c3

Please advise — this is a blocker for releasing my web app to production 🥲  Thank you!

(PS - I would just convert all my serverless functions to edge functions, but one of them uses libraries that aren't supported by the Edge runtime.)

**Reply by jessepinho** [reply]
@amyegan also, please let me know if I should create a new issue for this. I figured I should piggyback on this issue, since it seems like the exact same problem.

**Reply by amyegan** [reply]
Thanks for sharing these examples. I tried to reproduce the issue in my test app with Next 13.3.0, as specified in your project, and the example functions you provided. But both are working as expected for me. 😕 

Since it seems to be caused by some other interaction in the code and configuration options, please share a [minimal reproducible example](https://vercel.com/support/articles/creating-a-minimal-reproducible-example) without Firebase or other required keys needed. That will give us a better view of the problem 🙏

**Reply by jessepinho** [reply]
Hey @amyegan — thanks for getting back to me so quickly! I did indeed remove Firebase on my `dev` branch. Are you looking at my `main` branch? The minimal reproducible example is running on the `dev` branch (see [here](https://github.com/jessepinho-film-tools/film-tools-web/tree/dev/src/pages/api)).

Could it be that my serverless function is deployed on a Vercel server that's hanging? Is it possible to re-deploy my function to a different server or something? Thanks!

**Reply by amyegan** [reply]
It was the `buildCommand` in vercel.json that was throwing me off. 😅 Removed firebase there and got it to build.

When I run a preview deployment of this code and test the serverless function, it takes over 14 seconds which exceeds the Hobby plan function timeout [limit](https://vercel.com/docs/concepts/limits/overview#general-limits). I went into `src/pages/_app.page.tsx` and commented out the nested providers and containers so only `ThemeWrapper` was being returned. Then the function took under 3 seconds, which allows it to execute successfully.

You should take a look at the various provider wrappers included in `App` and consider modifying, relocating, or removing any you can. One or more of these appear to be the source of the problem.

**Reply by jessepinho** [reply]
@amyegan Thank you so much! I didn't realize stuff happening in the other pages could affect the serverless functions. This is a huge help 🙏🏻

**Reply by jessepinho** [reply]
@amyegan I just got the serverless function to work by removing all providers/etc. from my `_app.page.tsx`, so it looks like I have some investigating to do. Thanks again.

One question: does this mean that the function for generating the `_app` is on the same server as my serverless functions? And thus, if the `_app` hangs, the functions will too?

Thanks!

**Comment by NicmeisteR** [comment]
Experiencing something similar, been tinkering a lot and still can't get it working. https://github.com/orgs/vercel/discussions/1910
